### PR TITLE
DCS-839 Extracting out the booking request email into separate module

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: backend/services/emails/BookingRequest.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { RequestEmail, EmailSpec } from '../model'
+import { notifications } from '../../config'
+
+export default function BookingRequest(details: RequestEmail): EmailSpec {
```
Comment (andrewrlee) [inline backend/services/emails/BookingRequest.ts:4]: Each type of email would have one of these modules - create/ request/ delete  /update

File: backend/services/notificationService.ts [hunk 2]
```diff
@@ -20,41 +21,41 @@ export = class NotificationService {
...
 
-    const courtPersonalisation = { userName: name, ...personalisation }
-
-    this.sendEmail({
-      templateId: notifications.requestBookingCourtTemplateVLBAdminId,
-      email: vlb,
-      personalisation,
-    }).catch(error => {
-      log.error(`Failed to email the prison about a booking request: ${error.message}`, error.response?.data?.errors)
+  private async sendEmails(context: Context, username: string, spec: EmailSpec): Promise<void> {
```
Comment (andrewrlee) [inline backend/services/notificationService.ts:37]: one generic send method

File: backend/services/notificationService.ts [hunk 3]
```diff
@@ -20,41 +21,41 @@ export = class NotificationService {
...
 
-    this.sendEmail({
-      templateId: notifications.requestBookingCourtTemplateRequesterId,
-      email,
-      personalisation: courtPersonalisation,
-    }).catch(error => {
-      log.error(`Failed to email the requester a copy of the booking: ${error.message}`, error.response?.data?.errors)
-    })
+  public async sendBookingRequestEmails(context: Context, username: string, details: RequestEmail): Promise<void> {
+    await this.sendEmails(context, username, BookingRequest(details))
```
Comment (andrewrlee) [inline backend/services/notificationService.ts:58]: this is how you now send a set of emails of a type

File: backend/services/emails/BookingRequest.ts (hunk 1 above)
Comment (andrewrlee) [reply backend/services/emails/BookingRequest.ts:4]: (create will be coming in a follow up - moving down from the controller)

**Review by louisenorris — APPROVED** [review]
LGTM

**Pull request merged**